crates/storage_impl/src/database/store.rs

```diff
@@ -19,6 +22,7 @@ pub trait DatabaseStore: Clone + Send + Sync {
         config: Self::Config,
         tenant_config: &dyn TenantConfig,
         test_transaction: bool,
+        key_manager_state: keymanager::KeyManagerState,
```

Comment:
This should be option, without `KeyManagerState` also a `DatabaseStore` can be created, this is not mandatory constraint